cache/in-memory/src/event/voice_state.rs

```diff
@@ -19,25 +19,52 @@ impl InMemoryCache {
 
         // Check if the user is switching channels in the same guild (ie. they already have a voice state entry)
         if let Some(voice_state) = self.voice_states.get(&(guild_id, user_id)) {
-            if let Some(channel_id) = voice_state.channel_id() {
-                let remove_channel_mapping = self
-                    .voice_state_channels
-                    .get_mut(&channel_id)
-                    .map(|mut channel_voice_states| {
-                        channel_voice_states.remove(&(guild_id, user_id));
-
-                        channel_voice_states.is_empty()
-                    })
-                    .unwrap_or_default();
-
-                if remove_channel_mapping {
-                    self.voice_state_channels.remove(&channel_id);
-                }
+            let remove_channel_mapping = self
+                .voice_state_channels
+                .get_mut(&voice_state.channel_id())
+                .map(|mut channel_voice_states| {
+                    channel_voice_states.remove(&(guild_id, user_id));
+
+                    channel_voice_states.is_empty()
+                })
+                .unwrap_or_default();
+
+            if remove_channel_mapping {
+                self.voice_state_channels.remove(&voice_state.channel_id());
             }
         }
 
-        // Check if the voice channel_id does not exist, signifying that the user has left
-        if voice_state.channel_id.is_none() {
+        if let Some(channel_id) = voice_state.channel_id {
+            let cached_voice_state = CachedVoiceState {
+                channel_id,
+                deaf: voice_state.deaf,
+                guild_id,
+                mute: voice_state.mute,
+                request_to_speak_timestamp: voice_state.request_to_speak_timestamp,
+                self_deaf: voice_state.self_deaf,
+                self_mute: voice_state.self_mute,
+                self_stream: voice_state.self_stream,
+                self_video: voice_state.self_video,
+                session_id: voice_state.session_id,
+                suppress: voice_state.suppress,
+                token: voice_state.token,
+                user_id,
+            };
```

Comment:
I would like to extract initializers, so I would like to place this on `CachedVoiceState`. It doesn't have to be a public API